tests/integration/callbacks/test_pattern_matching.py

```diff
@@ -0,0 +1,167 @@
+from selenium.webdriver.support.select import Select
+
+all_app = """
+library(dash)
+library(dashHtmlComponents)
+library(dashCoreComponents)
+
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+
+app <- Dash$new()
+
+app$layout(htmlDiv(list(
+  htmlButton("Add Filter", id="add-filter", n_clicks=0),
+  htmlDiv(id="dropdown-container", children=list()),
+  htmlDiv(id="dropdown-container-output")
+)))
+
+
+app$callback(
+  output(id="dropdown-container", property = "children"),
+  params = list(
+    input(id = "add-filter", property = "n_clicks"),
+    state(id = "dropdown-container", property = "children")
+  ),
+  display_dropdowns <- function(n_clicks, children){
+    new_dropdown = dccDropdown(
+      id=list(
+        "index" = n_clicks,
+        "type" = "filter-dropdown"
+      ),
+      options = lapply(c("NYC", "MTL", "LA", "TOKYO"), function(x){
+        list("label" = x, "value" = x)
+      })
+    )
+    children[[n_clicks + 1]] <- new_dropdown
+    return(children)
+  }
+)
+
+
+app$callback(
+  output(id="dropdown-container-output", property="children"),
+  params = list(
+    input(id=list("index" = dash:::ALL, "type" = "filter-dropdown"), property= "value")
+  ),
+  display_output <- function(test){
+    ctx <- app$callback_context()
+    return(htmlDiv(
+      lapply(1:length(test), function(x){
+        return(htmlDiv(sprintf("Dropdown %s = %s", x, test[[x]])))
+      })
+    ))
+  }
+)
+
+
+app$run_server(debug=T)
+"""
+
+allsmaller_app = """
+
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+
+
+df <- read.csv('https://raw.githubusercontent.com/plotly/datasets/master/gapminder2007.csv', stringsAsFactors = F)
+
+
+app <- Dash$new()
+
+app$layout(htmlDiv(list(
+  htmlButton("Add Filter", id = "add-filter-ex3", n_clicks = 0),
+  htmlDiv(id = "container-ex3", children = list())
+)))
+
+
+app$callback(
+  output('container-ex3', 'children'),
+  params = list(
+    input('add-filter-ex3', 'n_clicks'),
+    state('container-ex3', 'children')
+  ),
+  display_dropdowns <- function(n_clicks, existing_children){
+    new_children <- htmlDiv(list(
+      dccDropdown(
+        id = list("index" = n_clicks, "type" = "filter-dropdown-ex3"),
+        options = lapply(unique(df$country), function(x){
+          list("label" = x, "value" = x)
+        }),
+        value = unique(df$country)[n_clicks + 1]
+      ),
+      htmlDiv(id = list("index" = n_clicks, "type" = "output-ex3"), children = list(unique(df$country)[n_clicks + 1]))
+    ))
+    
+    existing_children <- c(existing_children, list(new_children))
+  }
+)
+
+
+app$callback(
+  output(id = list("type" = "output-ex3", "index" = dash:::MATCH), property = "children"),
```

Comment:
I was surprised to see `dash:::MATCH` since we explicitly exported these symbols in the `.Rd`, i.e. there are export lines:

https://github.com/plotly/dashR/blob/2d84a88d6182b316bea8a0afb7eadc4d566e7d2e/R/dependencies.R#L82-L92

There's an extra step I should have mentioned before, to trigger a `roxygen2` update to the `NAMESPACE` file:

```r
Rscript -e 'devtools::document()'
```

I went ahead and executed this step, then committed the updated documentation, so now the objects are exported and we can do instead:

```suggestion
  output(id = list("type" = "output-ex3", "index" = MATCH), property = "children"),
```

It turns out that CRAN won't allow the `:::` syntax in submitted packages, but no sweat, we're all set there now.